test/controllers/products_controller_test.rb

```diff
@@ -2,13 +2,23 @@
 
 describe ProductsController do
   describe "index" do
-    it "should get index" do
+    it "successfully shows index" do
       # Act
       get products_path
 
       # Assert
       must_respond_with :success
     end
+
+    it "successfully shows index with no products" do
+      Product.all do |product|
+        product.destroy
+      end
+
```

Comment:
chris has made comments about using Product.destroy_all rather than looping. but it works the same. just wanted to share.